source/upgrade/downgrading-mattermost-server.rst

```diff
@@ -1,84 +1,99 @@
 Downgrading Mattermost Server
 =============================
 
-In most cases you can downgrade Mattermost Server using the same steps as :doc:`upgrading-mattermost-server`. The binaries can be found in the :doc:`version-archive`. We do not recommend downgrading more than one version back from your current installation.
-
-Downgrade from 5.0 to 4.10
----------------------------
-
-During the version 5.0 upgrade process, a migration is run to enable new features of the `advanced permissions system <https://docs.mattermost.com/onboard/advanced-permissions.html>`__. This changes the database in ways that result in it no longer being compatible with Mattermost server 4.10. If you need to downgrade from 5.0 or later to 4.10, it is necessary to follow the steps below.
+|all-plans| |self-hosted|
 
-You should gather the following information before starting the upgrade:
+.. |all-plans| image:: ../images/all-plans-badge.png
+  :scale: 30
+  :target: https://mattermost.com/pricing
+  :alt: Available in Mattermost Free and Starter subscription plans.
 
-Existing install directory - *{install-path}*
-  If you don't know where Mattermost Server is installed, use the ``whereis mattermost`` command. The output should be similar to */opt/mattermost/bin/mattermost*. The install directory is everything before the last occurrence of the string */mattermost*. In this example, the *{install-path}* is ``/opt``.
-Location of your local storage directory
-  The local storage directory contains all the files that users have attached to their messages. If you don't know its location, open the System Console and go to **Files > Storage** and read the value in **Local Storage Directory**. Relative paths are relative to the ``mattermost`` directory. For example, if the local storage directory is ``./data/`` then the absolute path is ``{install-path}/mattermost/data``.
-Owner and group of the install directory - *{owner}* and *{group}*
-  Use the ``ls -l {install-path}/mattermost/bin/mattermost`` command to get the owner and group.
+.. |self-hosted| image:: ../images/self-hosted-badge.png
+  :scale: 30
+  :target: https://mattermost.com/deploy
+  :alt: Available for Mattermost Self-Hosted deployments.
 
-#. In a terminal window on the server that hosts Mattermost Server, change to your home directory. If any, delete files and directories that might still exist from a previous download.
-
-   .. code-block:: sh
-
-     cd ~
+In most cases you can downgrade Mattermost Server using the same steps as :doc:`upgrading-mattermost-server`. The binaries can be found in the :doc:`version-archive`. We do not recommend downgrading more than one version back from your current installation.
 
-#. Stop Mattermost server.
+Downgrading from v6.0 to v5.38
```

Comment:
What about 6.0 to another version below 5.38? Are there more general instructions between minor versions (e.g. 5.38 -> 5.35)

Reply:
@streamer45 ^ Are we supporting additional downgrade paths beyond what's currently documented?

Reply:
We generally supported jumping through minor versions when upgrading. This is a bit different as it's a major one but it should technically work although there's always a risk attached as more changes go in and 6.0 is already a very heavy migration. Maybe we could offer a 5.x ESR to 6.x ESR official upgrade/downgrade path? Not sure what 6.x version will be ESR but it could be a good option. 

Slightly unrelated but the original document was made before 5.39 was planned but as that doesn't include any db schema change downgrading queries are unchanged and will equally work to both versions (5.38 and 5.39).

Reply:
@wiersgallak - Thoughts on above?

Reply:
I think the 6.x to 5.x ESR upgrade/downgrade would be a great addition.  We can add this later as our next ESR would be in January.